### DISCUSSION TITLE
Primefaces datatable selection on page refresh

### DISCUSSION THREAD

**Discussion by trawat** [primefaces/community#552] [PrimeFaces] [answered]
Having trouble clearing the data table selection. Using PF11 on Wildfly 11 with JDK 8.

I have a page - data table with multiple selection enabled. Backing bean is in Session scope. And, a split button to apply/ change the status of selection record.

<img width="1140" alt="Screenshot 2023-11-24 at 11 32 49 PM" src="https://github.com/primefaces/community/assets/17884127/27cf7daa-8f94-410f-af74-e5f8213f672d">

Above screenshot shows the data table and selection. Now, on changing the status below is how the datatable looks

<img width="1089" alt="Screenshot 2023-11-25 at 12 16 29 AM" src="https://github.com/primefaces/community/assets/17884127/5b529f29-4289-4c5f-955d-fa9b00884245">

Now, if I search _**Trolly**_ type, then my page looks like this

<img width="1178" alt="Screenshot 2023-11-25 at 12 17 57 AM" src="https://github.com/primefaces/community/assets/17884127/323047b6-d448-413d-9ecb-cfcef2239d25">

This is on PF11. Now, if I change the PF version to 12 then the above 2 cases/ issues are gone. 

**Third case** on _PF12_ - after changing the status if I use browser refresh once then the datatable starts showing the previously selected row and if I refresh again the split button also starts showing _row selected_ message.

Have created a project on github with all the files I am using - https://github.com/trawat/mvce. Do let me know in case I need to copy paste the relevant code in this discussion itself.

Need a way to avoid this behaviour of datatable on page refresh.

Have asked the same on [SO](https://stackoverflow.com/questions/77545331/primefaces-datatable-selection-on-page-refresh) as well.

**Comment by melloware** [comment]
Can you use this to make it runnable for us: https://github.com/primefaces/primefaces-test

**Reply by trawat** [reply]
@melloware here is the link - https://github.com/trawat/primefaces-dt-selection

In case, we need any more info, do let me know. It's the third case I am after. Thanks.

**Reply by melloware** [reply]
OK running your reproducer.  Can you be clear on the exact steps as I have no idea what I am looking at?
Like..

1. Filter by "Trolley".
2. Click on ...
3. Refresh the browser
4. ...

**Reply by trawat** [reply]
**Third case** - as other two works fine if I bump the version from PF 11 to 12.

Step - 

1. select row having trolley type and then use the split button to update the status to Open. 
2. Now, use browser refresh
3. Repeat step 2 again.

**Reply by melloware** [reply]
OK the issue is `multiviwState="true"` its remembering the selection from the last AJAX request when you press F5 refresh on the browser.  Not sure if its a bug or not with `multiviwState="true"` ??

**Reply by trawat** [reply]
Thanks for valuable input. Are there any steps or guideline to raise this as a bug ? or anything that might help to change this behaviour.

We need `multiViewState` attribute on our pages.

**Reply by melloware** [reply]
I would guess it is some kind of bug you could report it on PrimeFaces GitHub Issue and be sure to include your reproducer.

**Reply by melloware** [reply, marked as answer]
Fixed in 14.0.0 https://github.com/primefaces/primefaces/issues/11146
